docs/protocol/on-chain/channel-lifecycle.mdx

```diff
@@ -0,0 +1,375 @@
+   - Call the `resize` function with:
+     - The <Tooltip content="A unique identifier for a channel, computed as the hash of the channel configuration.">channelId</Tooltip> (remains unchanged)
+     - A candidate <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">State</Tooltip> with:
+       - `intent` = `StateIntent.RESIZE` (2)
+       - `version` = precedingState.version + 1
+       - `data` = ABI-encoded `int256[]` containing delta amounts (positive for deposit, negative for withdrawal)
+       - `allocations` = final Allocation[] after resize (absolute amounts)
+       - Signatures from **ALL participants** (consensus required)
+     - An array of proof States supporting the candidate
+
+2. The contract MUST:
+   - Verify the <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> is in ACTIVE status
+   - Verify all participants have signed the resize state
+   - Decode delta amounts from `candidate.data`
+   - Validate adjudicator approves the preceding state
+   - For positive deltas: Lock additional funds from custody account
+   - For negative deltas: Unlock funds back to custody account
+   - Update expected deposits to match new allocations
+   - Emit `Resized(channelId, deltaAllocations)` event
+
+3. The <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip>:
+   - **channelId remains UNCHANGED** (same channel persists)
+   - Status remains **ACTIVE** throughout
+   - Version increments by 1
+   - No new channel is created
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant Contract
+    
+    User->>Contract: resize(channelId, resizeState, proofs)
+    Note right of Contract: Same channelId<br/>State version + 1<br/>Intent = RESIZE
+    Contract->>Contract: Verify signatures (all participants)
+    Contract->>Contract: Decode delta amounts from state.data
+    Contract->>Contract: Lock funds (positive deltas)
+    Contract->>Contract: Unlock funds (negative deltas)
+    Contract->>Contract: Update expected deposits
+    Contract->>User: Resized(channelId, deltas)
+    Note right of User: Same channelId<br/>Channel still ACTIVE
+```
+
+**Use Cases**:
+- Increasing channel capacity (positive delta: adding funds)
+- Decreasing channel capacity (negative delta: removing funds)
+- Adjusting allocations while maintaining channel continuity
+
+:::tip In-Place Update
+The resize operation updates the channel **in place**. The channelId stays the same, and the channel remains ACTIVE throughout. This differs from closing and reopening, which would create a new channel.
+:::
+## State Transition Summary
+
+The complete channel lifecycle state machine:
+
+```mermaid
```

Comment:
You should also update the diagram and the "valid transitions" table with:
- an arrow from "void" to "active" with "create(...) with signatures from all participants"
- resize(...) with arrow from "active" to "active"

Reply:
[f889dd9d34e1744c4d0d2598c3c590f87acb676b](https://github.com/layer-3/docs/pull/69/commits/f889dd9d34e1744c4d0d2598c3c590f87acb676b)